tests/syft/grid/duet_test.py

```diff
@@ -1,60 +1,39 @@
-# import syft as sy
-# from syft.grid.duet.request import RequestResponse, RequestMessage, RequestStatus
-# from syft.core.common import UID
-# import torch as th
+import syft as sy
+import torch as th
 
 
-# def test_request_message_creation() -> None:
-#     obj = RequestMessage(
-#         request_name="request", request_description="request description"
-#     )
+def test_duet_send_and_get() -> None:
+    duet = sy.Duet(host="127.0.0.1", port=5001)
 
-#     assert obj.request_name == "request"
-#     assert obj.request_description == "request description"
+    x = th.tensor([1, 2, 3])
+    xp = x.send(duet)
 
+    assert xp.id_at_location == x.id
 
-# def test_request_message_serde() -> None:
-#     obj = RequestMessage(
-#         request_name="request", request_description="request description"
-#     )
-#     serialized_request = sy.serialize(obj=obj)
-#     new_obj = sy.deserialize(blob=serialized_request)
+    yp = xp + xp
 
-#     assert obj.request_name == new_obj.request_name
-#     assert obj.request_description == new_obj.request_description
-#     assert obj.request_id == new_obj.request_id
+    y = yp.get()
+    assert ((x + x) == y).all()
 
+    duet.stop()
 
-# def test_request_response() -> None:
-#     id = UID()
-#     status = RequestStatus.Pending
-#     obj = RequestResponse(status=status, request_id=id)
 
-#     assert obj.status == status
-#     assert obj.request_id == id
+def test_duet_searchable_functionality() -> None:
+    duet = sy.Duet(host="127.0.0.1", port=5001)
 
+    xp = th.tensor([1, 2, 3]).tag("some", "diabetes", "data").send(duet)
+    xp2 = (
+        th.tensor([1, 2, 3]).tag("some", "diabetes", "data").send(duet, searchable=True)
+    )
 
-# def test_request_response_serde() -> None:
-#     obj = RequestResponse(status=RequestStatus.Pending, request_id=UID())
+    guest = sy.Duet(domain_url="http://127.0.0.1:5001/")
 
-#     serialized_obj = sy.serialize(obj=obj)
-#     new_obj = sy.deserialize(blob=serialized_obj)
+    assert len(guest.store) == 1
+    assert len(duet.store) == 2
 
-#     assert obj.status == new_obj.status
-#     assert obj.request_id == new_obj.request_id
+    assert guest.store[0].id_at_location == xp2.id_at_location
 
+    del xp
+    del xp2
 
-# def test_duet_send_and_get() -> None:
-#     duet = sy.Duet(host="127.0.0.1", port=5001)
-
-#     x = th.tensor([1, 2, 3])
-#     xp = x.send(duet)
-
-#     assert xp.id_at_location == x.id
-
-#     yp = xp + xp
-
-#     y = yp.get()
-#     assert ((x + x) == y).all()
-
-#     duet.stop()
+    duet.stop()
```

Comment:
nit: Maybe we could use a pytest feature (I don't remember how it's called right now) to instantiate and clean the Duet for each test (this can be done later, of course!)

Reply:
Agreed - we only have 2 duet tests for now so i think we can cleanup later. But if you know how to do this that'd be awesome.